seatunnel-connectors/seatunnel-connectors-spark/seatunnel-connector-spark-http/src/main/java/org/apache/seatunnel/spark/source/Http.java

```diff
@@ -0,0 +1,142 @@
+        String url = config.getString(Settings.SOURCE_HTTP_URL);
+        String method = config.hasPath(Settings.SOURCE_HTTP_METHOD) ?
+            config.getString(Settings.SOURCE_HTTP_METHOD) : GET;
+        String header = config.hasPath(Settings.SOURCE_HTTP_HEADER) ?
+            config.getString(Settings.SOURCE_HTTP_HEADER) : "";
+        String requestParams = config.hasPath(Settings.SOURCE_HTTP_REQUEST_PARAMS) ?
+            config.getString(Settings.SOURCE_HTTP_REQUEST_PARAMS) : "";
+        String syncPath = config.hasPath(Settings.SOURCE_HTTP_SYNC_PATH) ?
+            config.getString(Settings.SOURCE_HTTP_SYNC_PATH) : "";
+
+        JavaSparkContext jsc = JavaSparkContext.fromSparkContext(spark.sparkContext());
+
+        Map requestMap = jsonToMap(requestParams);
+        String syncValues = getSyncValues(jsc, syncPath);
+        LOG.info("sync values->{}", syncValues);
+        Map syncMap = jsonToMap(syncValues);
+        if (!syncMap.isEmpty()) {
+            requestMap.putAll(syncMap);
+        }
+
+        HttpClientResult response = new HttpClientResult();
+        try {
+            Map headerMap = jsonToMap(header);
+            if (POST.equals(method)) {
+                response = HttpClientUtils.doPost(url, headerMap, requestMap);
+            } else {
+                response = HttpClientUtils.doGet(url, headerMap, requestMap);
+            }
+        } catch (Exception e) {
+            LOG.error("http call error!", e);
+        }
+
+        LOG.info("http respond code->{}", response.getCode());
+
+        List<String> array = new ArrayList<>();
+        array.add(response.getContent());
+        JavaRDD<String> javaRDD = jsc.parallelize(array);
+        DataFrameReader reader = spark.read().format("json");
+        return reader.json(javaRDD);
+    }
+
+    private String getSyncValues(JavaSparkContext jsc, String syncPath) {
+        if (null == syncPath || syncPath.isEmpty()) {
+            return "";
+        }
+        Configuration hadoopConf = jsc.hadoopConfiguration();
+        List<String> values = new ArrayList<>();
+        try {
+            FileSystem fs = FileSystem.get(hadoopConf);
+            Path path = new Path(syncPath);
+            boolean exists = fs.exists(path);
+            if (exists) {
+                JavaRDD<String> checkPoint = jsc.textFile(syncPath);
+                values = checkPoint.collect();
+
+            }
+        } catch (IOException e) {
+            LOG.error("getSyncValues error, syncPath is {}", syncPath, e);
+        }
+        return values.isEmpty() ? "" : values.iterator().next();
+    }
+
+    private Map jsonToMap(String content) {
+        Map map = new HashMap<>(INITIAL_CAPACITY);
+        if (null == content || content.isEmpty()) {
+            return map;
+        }
+
+        try {
+            return new ObjectMapper().readValue(content, HashMap.class);
+        } catch (IOException e) {
+            LOG.error("{} json to map error!", content, e);
```

Comment:
Why catch this exception, make this case to be normal? I think in most case, if you think this exception just need to be resolved in this method, you need to add a comment why this exception needn't to throw.

Reply:
Catching this exception only records the log, does not handle it, and does not affect the main process.

Reply:
You need to add comment when you catch a exception and kill the exception in this catch.